server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1248,28 +1323,45 @@ private void logPools(Collection<Pool> pools) {
         return enforced;
     }
 
+    private Map<String, Integer> convertToMap(List<PoolQuantity> poolQuantities) {
+        Map<String, Integer> result = new HashMap<String, Integer>();
+        for (PoolQuantity poolQuantity : poolQuantities) {
+            result.put(poolQuantity.getPool().getId(), poolQuantity.getQuantity());
+        }
+        return result;
+    }
+
     /**
-     * Request an entitlement by pool.. If the entitlement cannot be granted,
-     * null will be returned. TODO: Throw exception if entitlement not granted.
-     * Report why.
+     * Request an entitlement by poolid and quantity
      *
      * @param consumer consumer requesting to be entitled
-     * @param pool entitlement pool to consume from
-     * @return Entitlement
+     * @param poolQuantities a map of entitlement pool ids and the respective
+     *        quantities to consume from
+     * @return Entitlements A list of entitlements created if the request is
+     *         successful
      * @throws EntitlementRefusedException if entitlement is refused
      */
     @Override
-    public Entitlement entitleByPool(Consumer consumer, Pool pool,
-        Integer quantity) throws EntitlementRefusedException {
-        return addOrUpdateEntitlement(consumer, pool, null, quantity,
-            false, CallerType.BIND);
+    @Transactional
+    public List<Entitlement> entitleByPools(Consumer consumer, Map<String, Integer> poolQuantities)
+        throws EntitlementRefusedException {
+        if (MapUtils.isNotEmpty(poolQuantities)) {
+            return addOrUpdateEntitlements(consumer, poolQuantities, null, false, CallerType.BIND);
+        }
+        return new ArrayList<Entitlement>();
     }
 
     @Override
-    public Entitlement ueberCertEntitlement(Consumer consumer, Pool pool, Integer quantity)
+    public Entitlement ueberCertEntitlement(Consumer consumer, Pool pool)
         throws EntitlementRefusedException {
-
-        return addOrUpdateEntitlement(consumer, pool, null, 1, true, CallerType.UNKNOWN);
+        Map<String, Integer> poolQuantities = new HashMap<String, Integer>();
+        poolQuantities.put(pool.getId(), 1);
+        List<Entitlement> result = addOrUpdateEntitlements(consumer, poolQuantities, null, true,
+                CallerType.UNKNOWN);
+        if (CollectionUtils.isNotEmpty(result)) {
+            return result.get(0);
```

Comment:
A comment here would be helpful.  I'm a little unclear why only the first element in the list is being returned.  I.e. is the list always of length one or are all the elements in the list going to be equal?